src/Compilers/CSharp/Test/Emit/CodeGen/NullCoalescingAssignmentTests.cs

```diff
@@ -82,41 +85,57 @@ .locals init (int? V_0)
 }
 ");
 
+            // When the optimizer is on, the local is entirely elided as the result of the assignment isn't used again
```

Comment:
That's technically not what's happening here. The optimizer is turning the local into a stack local. It's being used, but since there are no intervening stack requirements, we can just keep the value on the stack instead of turning it into a local.

Reply:
Thanks for the better explanation. I'll update the comment in a follow up

---
In reply to: [203168514](https://github.com/dotnet/roslyn/pull/28598#discussion_r203168514) [](ancestors = 203168514)